src/cmd-build

```diff
@@ -79,10 +92,36 @@ if [ -f "${changed_stamp}" ]; then
     cp -a --reflink=auto ${composejson} ${workdir}/tmp/compose-${commit}.json
 else
     commit=${previous_commit}
-    # Grab the previous JSON
-    cp -a --reflink=auto ${workdir}/tmp/compose-${previous_commit}.json ${composejson}
+    image_input_checksum=$((echo ${commit} && echo ${kickstart_checksum}) | sha256sum_str)
+    if [ "${image_input_checksum}" = "${previous_image_input_checksum}" ]; then
+        echo "No changes in image inputs."
+        exit 0
+    fi
+
+    # Grab the previous treecompose JSON (local developer case: treecompose succeeded but
+    # image build failed) if possible, otherwise grab the previous build
+    cached_previous_composejson=${workdir}/tmp/compose-${previous_commit}.json
+    if [ -f "${cached_previous_composejson}" ]; then
+        cp -a --reflink=auto ${cached_previous_composejson} ${composejson}
+    else
+        if [ -z "${previous_build}" ]; then
+            fatal "compose tree had no changes, but no previous build or cached data"
+        fi
+        # This will have all of the data from the previous build, but we'll
+        # overwrite things.
+        cp -a --reflink=auto ${previous_builddir}/meta.json ${composejson}
+    fi
 fi
+
+image_input_checksum=$((echo ${commit} && echo ${kickstart_checksum}) | sha256sum_str)
```

Comment:
we are calcualating `$image_input_checksum` twice. Here and on line 95. Is that intended?